src/utils/scheduler.h

```diff
@@ -46,5 +46,40 @@ class Scheduler {
     std::vector<PeriodTask> period_tasks_;
 };
 
+class Timer {
+public:
+    Timer(uint32_t _duration, std::function<void()>&& f) : duration_(_duration), f_(std::move(f)), stop_(true) {}
+    ~Timer() {
+        Stop();
+    }
+
+    void Reset() {
+        if (duration_ == 0) {
+            return;
```

Comment:
consider adding a log here to indicate the invalid duration

Reply:
so I will add a log in the constructor